File: single-language-analysis/dataset-files/java/TestSmell106.java
import org.junit.Test;
import static org.junit.Assert.*;

public class TestSmell106 {

    @Test
    public void testOnlyOnErrors() {
        int x = 2;
        int callCount = 0;
        boolean raised = false;
        int n = 0;

        Cache cache = new Cache(onlyOnErrors = new Class[]{ZeroDivisionError.class, ConnectionError.class});
        
        Runnable fn = new Runnable() {
            public int run() {
                callCount++;
                try {
                    return 0 / x;
                } catch (ZeroDivisionError e) {
                    raised = true;
                    throw e;
                } finally {
                    n++;
                    x--;
                }
            }
        };

        assertEquals(1, fn.run());
        assertEquals(1, callCount);
        assertEquals(2, fn.run());
        assertEquals(2, callCount);
        assertEquals(0, x);
        assertEquals(2, fn.run());
        assertEquals(3, callCount);
        assertTrue(raised);
        assertEquals(2, fn.run());
        x = 1;
        raised = false;
        assertEquals(3, fn.run());
        assertEquals(5, callCount);
        assertEquals(3, fn.run());
        assertTrue(raised);
        assertEquals(6, callCount);
        fn._cache.clear();
        try {
            fn.run();
            fail("Expected ZeroDivisionError");
        } catch (ZeroDivisionError e) {
            // Expected exception
        }
    }
}

File: single-language-analysis/dataset-files/python/test_smell_106.py
# Original URL: https://github.com/bofm/python-caching/blob/16de50c119f8896e923e3f10488ecf6ffe535f45/tests/test_cache.py#L330-L368

def test_smell_106():

    x = 2
    call_count = 0
    raised = False
    n = 0

    @Cache(only_on_errors=(ZeroDivisionError, ConnectionError))
    def fn():
        nonlocal x, n, call_count, raised
        call_count += 1
        try:
            0 / x
        except ZeroDivisionError:
            raised = True
            raise
        n += 1
        x -= 1
        return n

    assert fn() == 1
    assert call_count == 1
    assert fn() == 2
    assert call_count == 2
    assert x == 0
    assert fn() == 2  # ZeroDivision and returning cached result
    assert call_count == 3
    assert raised
    assert fn() == 2
    x = 1
    raised = False
    assert fn() == 3
    assert call_count == 5
    assert fn() == 3  # ZeroDivision and returning cached result
    assert raised
    assert call_count == 6
    fn._cache.clear()
    with pytest.raises(ZeroDivisionError):
        fn()